GitUI/CommandsDialogs/FormPush.cs

```diff
@@ -737,24 +750,7 @@ private void UpdateRemoteBranchDropDown()
 
             if (_selectedRemote is not null)
             {
-                foreach (IGitRef head in GetRemoteBranches(_selectedRemote.Name))
-                {
-                    if (_NO_TRANSLATE_Branch.Text != head.LocalName)
-                    {
-                        RemoteBranch.Items.Add(head.LocalName);
-                    }
-                }
-
-                HashSet<string> remoteBranchesSet = GetRemoteBranches(_selectedRemote.Name).Select(b => b.LocalName).ToHashSet();
-                IEnumerable<IGitRef> onlyLocalBranches = GetLocalBranches().Where(b => !remoteBranchesSet.Contains(b.LocalName));
```

Comment:
This could maybe add a value if the local has no tracking remote and you want to add one.
But better to remove, less confusing to require a manual entering then to see a lot of branches not related to the remote
I miss this in the commit message though

Reply:
This part has been removed so the remote combobox won't be filled with a all the local branches names anymore. So you won't "see a lot of branches not related to the remote" anymore.

>This could maybe add a value if the local has no tracking remote and you want to add one.

🤔 It's what I did...
One value will only be added if you select a not tracked local branch. So not a lot of new values added.

>less confusing to require a manual entering

Not sure if I get you because if you mean that you want to oblige the user to type completely the remote branch name himself, not sure I want that because it will be a big UX regression because most of the times you just want to push to a remote with the same name (it's even the best practice). For me it's good enough if the user has to tweak a little the pre-selected value if not 100% accurate....